src/main/java/org/prebid/server/bidder/connatix/ConnatixBidder.java

```diff
@@ -94,6 +104,39 @@ public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request
         return Result.of(httpRequests, errors);
     }
 
+    private String getOptimalEndpointUrl(BidRequest request) {
+        try {
+            final URIBuilder uriBuilder = new URIBuilder(endpointUrl);
+            return getUserId(request)
+                    .map(userId -> getDataCenterCode(userId))
+                    .map(dataCenterCode -> uriBuilder.addParameter("dc", dataCenterCode))
+                    .orElse(uriBuilder)
+                    .build()
+                    .toString();
+        } catch (URISyntaxException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
```

Comment:
```
    private String getOptimalEndpointUrl(BidRequest request) {
        final Optional<String> dataCenterCode = getUserId(request)
                .map(ConnatixBidder::getDataCenterCode);

        if (dataCenterCode.isEmpty()) {
            return endpointUrl;
        }

        try {
            return new URIBuilder(endpointUrl)
                    .addParameter("dc", dataCenterCode.get())
                    .build()
                    .toString();
        } catch (URISyntaxException e) {
            throw new PreBidException(e.getMessage());
        }
    }
```

Reply:
I see that @AntoxaAntoxic asked you to change this piece of code. It seems a bit confusing to me (due to `.map(<toBuilder>)`), let's go back to the old version.

Please make this small change and we will merge the pr as soon as possible.

Reply:
@CTMBNara addressed!